### PR TITLE
Add offworld models to super metroid pickup db

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/games/super_metroid/pickup_database/pickup-database.json [hunk 1]
```diff
@@ -209,7 +249,9 @@
             "pickup_category": "visor",
             "broad_category": "visor",
             "model_name": "X-Ray Scope",
-            "offworld_models": {},
+            "offworld_models": {
+                "prime1": "X-Ray Visor"
```
Comment (duncathan) [inline randovania/games/super_metroid/pickup_database/pickup-database.json:253]: could add pulse radar for dread

File: randovania/games/super_metroid/pickup_database/pickup-database.json [hunk 2]
```diff
@@ -269,7 +330,9 @@
             "pickup_category": "morph_ball",
             "broad_category": "morph_ball_related",
             "model_name": "Spring Ball",
-            "offworld_models": {},
+            "offworld_models": {
+                "am2r": "sItemJumpBall",
```
Comment (duncathan) [inline randovania/games/super_metroid/pickup_database/pickup-database.json:334]: I might suggest using the morph model for prime and dread

File: randovania/games/super_metroid/pickup_database/pickup-database.json [hunk 3]
```diff
@@ -341,7 +428,12 @@
             "pickup_category": "energy_tank",
             "broad_category": "life_support",
             "model_name": "Reserve Tank",
-            "offworld_models": {},
+            "offworld_models": {
+                "am2r": "sItemEnergyTank",
+                "prime1": "Energy Tank",
+                "prime2": "EnergyTank",
+                "dread": "item_energytank"
```
Comment (duncathan) [inline randovania/games/super_metroid/pickup_database/pickup-database.json:435]: if it's desirable to distinguish between etanks and reserve tanks, echoes can use EnergyTankSmall and dread can use energy parts

**Review by duncathan — CHANGES_REQUESTED** [review]

File: randovania/games/super_metroid/pickup_database/pickup-database.json [hunk 4]
```diff
@@ -107,7 +112,11 @@
             "pickup_category": "beam",
             "broad_category": "beam_related",
             "model_name": "Ice Beam",
-            "offworld_models": {},
+            "offworld_models": {
+                "am2r": "sItemIceBeam",
+                "prime1": "Ice Beam",
+                "dread": "powerup_icemissile"
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/super_metroid/pickup_database/pickup-database.json:118]: Fair enough.

File: randovania/games/super_metroid/pickup_database/pickup-database.json (hunk 2 above)
Comment (Miepee) [reply randovania/games/super_metroid/pickup_database/pickup-database.json:334]: This was copied from am2r's implementation, should I go add morph model there too then?

File: randovania/games/super_metroid/pickup_database/pickup-database.json (hunk 3 above)
Comment (Miepee) [reply randovania/games/super_metroid/pickup_database/pickup-database.json:435]: Was not aware those existed. I assume prime1 doesn't have an equivalent there?

Comment (duncathan) [reply randovania/games/super_metroid/pickup_database/pickup-database.json:435]: not yet, no

**Review by duncathan — APPROVED** [review]

**Pull request merged**